TrafficCapture/captureOffloader/src/main/java/org/opensearch/migrations/trafficcapture/InMemoryConnectionCaptureFactory.java

```diff
@@ -0,0 +1,54 @@
+package org.opensearch.migrations.trafficcapture;
+
+import com.google.protobuf.CodedOutputStream;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+import java.io.ByteArrayOutputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.WeakHashMap;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.BiFunction;
+
+public class InMemoryConnectionCaptureFactory implements IConnectionCaptureFactory {
```

Comment:
It may be helpful to call out this is only used for testing or specify if you are planning an additional purpose here